frontend/src/components/pages/CreateReview/BookModal.tsx

```diff
@@ -0,0 +1,374 @@
+import AddNumberInput from "./AddNumberInput";
+import AddSelect from "./AddSelect";
+import AddStringInput from "./AddStringInput";
+import AddStringInputList from "./AddStringInputList";
+
+interface BookModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  booksAdded: Book[];
+  handleBooksAdded: (b: Book[]) => void;
+}
+
+const kStartingYear = 1967;
+const kMinAge = 0;
+const kMaxAge = 150;
+const kMinPrice = 0;
+const kMaxPrice = 1000;
+
+/**
+ * Modal for user to input new book for review
+ * */
+const BookModal = (props: BookModalProps): React.ReactElement => {
+  const { isOpen, onClose, booksAdded, handleBooksAdded } = props;
+
+  // required Book Fields
+  const [prefix, setPrefix] = useState<string>("");
+  const [title, setTitle] = useState<string>("");
+  const [authors, setAuthors] = useState<string[]>([]);
+  const [publisher, setPublisher] = useState<string>("");
+  const [publicationYear, setPublicationYear] = useState<string>("");
+  const [format, setFormat] = useState<string>("");
+  const [isbn, setIsbn] = useState<string>("");
+  const [price, setPrice] = useState<number>(0);
+  const [coverImage, setCoverImage] = useState<string>("");
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [genre, setGenre] = useState<string>("");
+  const [minAge, setMinAge] = useState<number>(0);
+  const [maxAge, setMaxAge] = useState<number>(0);
+
+  // optional book fields
+  const [seriesName, setSeriesName] = useState<string>("");
+  const [seriesOrder, setSeriesOrder] = useState<string>("");
+  const [illustrators, setIllustrator] = useState<string[]>([]);
+  const [translators, setTranslator] = useState<string[]>([]);
+
+  /** Builds array of years starting from current year */
+  const generateYearsArray = () => {
+    const max = new Date().getFullYear();
+    const years = [];
+
+    for (let i = kStartingYear; i <= max; i += 1) {
+      years.push(i.toString());
+    }
+    return years.reverse();
+  };
+
+  /** Clears book data */
+  const clearBookData = () => {
+    setTitle("");
+    setCoverImage("");
+    setPrefix("");
+    setSeriesOrder("");
+    setIllustrator([]);
+    setTranslator([]);
+    setFormat("");
+    setGenre("");
+    setMinAge(0);
+    setMaxAge(0);
+    setAuthors([]);
+    setPublisher("");
+    setSeriesName("");
+    setIsbn("");
+    setPrice(0);
+    setPublicationYear("");
+  };
+
+  // Calls clearBookData() whenever modal is closed
+  useEffect(() => {
+    if (isOpen === false) clearBookData();
+  }, [isOpen]);
+
+  /** Ensure that ISBN is valid or an empty field */
+  const isEmptyOrValidISBN = () => {
+    return (
+      isbn === "" || // necessary to ensure that empty form won't error
+      (isbn.length === 13 &&
+        !Number.isNaN(Number(isbn)) &&
+        Number.isInteger(Number(isbn)))
+    );
+  };
+
+  /** Check that all required modal fields are properly filled out */
+  const hasRequired = () => {
```

Comment:
I don't think these two need to be functions, might make sense for them to be boolean expressions, i.e.
```
const isEmptyOrValidISBN =
    isbn === "" ||
    (isbn.length === 13 &&
      !Number.isNaN(Number(isbn)) &&
      Number.isInteger(Number(isbn)));
```

Reply:
good call sir!